demos/common/mqtt_demo_helpers/mqtt_demo_helpers.c

```diff
@@ -233,13 +282,86 @@ static uint32_t prvGetTimeMs( void );
 
 /*-----------------------------------------------------------*/
 
+/*-----------------------------------------------------------*/
+
+static int32_t prvGenerateRandomNumber()
+{
+    const uint32_t ulMultiplier = 0x015a4e35UL, ulIncrement = 1UL;
+
+    ulNextRand = ( ulMultiplier * ulNextRand ) + ulIncrement;
+    return( ( int32_t ) ( ulNextRand >> 16UL ) & 0x7fffUL );
+}
+
+/*-----------------------------------------------------------*/
+
+static BaseType_t prvSeedRandomNumberGenerator()
```

Comment:
Where is this function invoked?

Reply:
Thanks for catching it. I have added a call-site for this function in this file in the `prvConnectToServerWithBackoffRetries` function.